draft-ppm-protocol.md

```diff
@@ -647,147 +676,222 @@ be cryptographically protected as described in {{helper-state}}.
 ~~~~
 Leader                                                 Helper
 
-AggregateReq (Reports 1-10) -------------------------------->  \
-<------------------------------------ AggregateResp (State 1)  | Reports
-AggregateReq (continued, State 1)      --------------------->  | 10-11
-<------------------------------------ AggregateResp (State 2)  /
+VerifyStartReq (Reports 1-10) -------------------------------->  \
+<------------------------------------ VerifyStartResp (State 1)  | Reports
+VerifyStartReq (continued, State 1) -------------------------->  | 10-11
+<------------------------------------ VerifyStartResp (State 2)  /
 
 
-AggregateReq (Reports 11-20, State 2) ---------------------->  \
-<------------------------------------ AggregateResp (State 3)  | Reports
-AggregateReq (continued, State 3) -------------------------->  | 20-21
-<------------------------------------ AggregateResp (State 4) /
+VerifyNextReq (Reports 11-20, State 2) ---------------------->  \
+<------------------------------------ VerifyNextResp (State 3)  | Reports
+VerifyNextReq (continued, State 3) -------------------------->  | 20-21
+<------------------------------------ VerifyNextResp (State 4)  /
 
 OutputShareReq (State 4) ----------------------------------->
 <----------------------------------- OutputShareResp (Result)
 ~~~~
 {: #pa-aggregate-flow title="Aggregation Process (batch size=20)"}
 
-[OPEN ISSUE: Should there be an indication of whether a given AggregateReq is a
-continuation of a previous sub-batch?]
-
 [TODO: Decide if and how the collector's request is authenticated.]
 
 
-### Aggregate Request
+### Verify-Start Request
 
-The AggregateReq request is used by the leader to send a set of reports to the
-helper. These reports MUST all be associated with the same PPM task. [[OPEN
-ISSUE: And the same batch, right?]]
+The *verify-start request* request is used by the leader to send a set of
+reports to the helper. These reports MUST all be associated with the same PPM
+task. [[OPEN ISSUE: And the same batch, right?]]
 
-For each aggregator endpoint `[aggregator]` in `AggregateReq.task_id`'s
-parameters except its own, the leader sends a POST request to
-`[aggregator]/aggregate` with the following message:
+For the second aggregator endpoint `[aggregator]` in `VerifyStartReq.task_id`'s
+parameters, the leader sends a POST request to `[aggregator]/aggregate_start`
+with the following message:
 
 ~~~
 struct {
   TaskID task_id;
   opaque helper_state<0..2^16>;
-  AggregateSubReq seq<1..2^24-1>;
-} AggregateReq;
+  opaque output_param<0..2^16-1>;
```

Comment:
Are there any risks with letting the leader insert arbitrary `output_param` into requests sent to helper? In the Priov3 case, it seems this should always be empty or absent (because the output param is already known to all the parties participating in a task).

In the Hits case, though, a leader could tamper with the string prefixes in the collector's queries. This might be beyond the scope of this particular PR, but maybe the helper and collector should establish a mutually authenticated channel using each other's HPKE configs to ensure leader can't tamper with the output param.

Reply:
In terms of soundness, there is a risk of a network attacker tampering with this. For that we'll need some sort of sender authentication in the leader<->helper channel. (We've discussed this type of threat before, but I'm not sure we have a ticket for it.)

In terms of privacy, mutual authentication doesn't buy us anything because the leader can still cheat. That said, privacy ought to hold here regardless of what the cheating leader does (so long as the helper itself is honest). Of course, we need to prove this for every VDAF we want to implement. (For Prio this is trivial since the only valid output parameter is "".)

Reply:
I wrote this up in #155. I think we should just put a TODO or a NOTE here referencing this issue and move on with this PR.